### PR TITLE
Fix listing update/httpput bugs by updating GenericRepository.UpdateAsync method. #81.

### PR DESCRIPTION
**Pull request by antoncarlgren**

### CODE REVIEW

File: src/GT.Core/Services/Impl/GTListingService.cs [hunk 1]
```diff
@@ -60,11 +60,11 @@ public class GTListingService : IGTListingService
...
-				if (signedInUserId is null)
-				{
-					_logger.LogWarning($"Attempted to add a new listing with a null reference on signedInUserId.");
-					return null;
-				}
+				//if (signedInUserId is null)
+				//{
+				//	_logger.LogWarning($"Attempted to add a new listing with a null reference on signedInUserId.");
+				//	return null;
+				//}
```
Comment (RasmusBroborg) [inline src/GT.Core/Services/Impl/GTListingService.cs:67]: Old impl needs to be kept.

File: src/GT.Core/Services/Impl/GTListingService.cs [hunk 2]
```diff
@@ -89,29 +89,44 @@ public class GTListingService : IGTListingService
 				{
 					if (!await _companyService.ExistsByNameAsync(listingDTO.Employer))
 					{
-						await _companyService.AddAsync(new CompanyDTO() { Name = listingDTO.Employer });
+						await _companyService
+							.AddAsync(new CompanyDTO() { Name = listingDTO.Employer });
 					}
-					entity.Employer = await _companyRepository.GetAll().Where(e => e.Name == listingDTO.Employer).FirstOrDefaultAsync();
+					var emp = await _companyRepository
```
Comment (RasmusBroborg) [inline src/GT.Core/Services/Impl/GTListingService.cs:95]: Var name should be employer as we have been rather explicit with variable names.

File: src/GT.Core/Services/Impl/GTListingService.cs [hunk 3]
```diff
@@ -89,29 +89,44 @@ public class GTListingService : IGTListingService
...
 				if (listingDTO.Location is not null)
 				{
 					if (!await _locationService.ExistsByNameAsync(listingDTO.Location))
 					{
-						await _locationService.AddAsync(new LocationDTO() { Name = listingDTO.Location });
+						await _locationService
+							.AddAsync(new LocationDTO() { Name = listingDTO.Location });
 					}
-					entity.Location = await _locationRepository.GetAll().Where(e => e.Name == listingDTO.Location).FirstOrDefaultAsync();
+					var loc = await _locationRepository
```
Comment (RasmusBroborg) [inline src/GT.Core/Services/Impl/GTListingService.cs:110]: Please use descriptive variable naming.

Comment (RasmusBroborg) [reply src/GT.Core/Services/Impl/GTListingService.cs:110]: Ex: 'location' instead of 'loc'

File: src/GT.Core/Services/Impl/GTListingService.cs [hunk 4]
```diff
@@ -89,29 +89,44 @@ public class GTListingService : IGTListingService
...
-						await _experienceLevelService.AddAsync(new ExperienceLevelDTO() { Name = listingDTO.ExperienceLevel });
+						await _experienceLevelService
+							.AddAsync(new ExperienceLevelDTO() { Name = listingDTO.ExperienceLevel });
 					}
-					entity.ExperienceLevel = await _experienceLevelRepository.GetAll().Where(e => e.Name == listingDTO.ExperienceLevel).FirstOrDefaultAsync();
+					var exp = await _experienceLevelRepository
+						.GetAll()
+						.FirstOrDefaultAsync(e => e.Name == listingDTO.ExperienceLevel);
+
+					entity.ExperienceLevel = exp;
```
Comment (RasmusBroborg) [inline src/GT.Core/Services/Impl/GTListingService.cs:129]: Descriptive.

**Review by RasmusBroborg — CHANGES_REQUESTED** [review]

File: src/GT.Core/Services/Impl/GTListingService.cs (hunk 1 above)
Comment (antoncarlgren) [reply src/GT.Core/Services/Impl/GTListingService.cs]: Fix in commit.

File: src/GT.Core/Services/Impl/GTListingService.cs (hunk 3 above)
Comment (antoncarlgren) [reply src/GT.Core/Services/Impl/GTListingService.cs]: Fix in commit.

**Review by RasmusBroborg — APPROVED** [review]

**Pull request merged**